examples/step-61/step-61.cc

```diff
@@ -520,87 +586,69 @@ namespace Step61
-    // to compute the error in the $L_2$ norm on each cell.
-    // Finally, we get the global $L_2$ norm.
     Vector<float> difference_per_cell(triangulation.n_active_cells());
-    VectorTools::integrate_difference(interior_dof_handler,
-                                      interior_solution,
-                                      Solution<dim>(),
+    const ComponentSelectFunction<dim> select_interior_pressure(0, 2);
+    VectorTools::integrate_difference(dof_handler,
+                                      solution,
+                                      ExactPressure<dim>(),
                                       difference_per_cell,
                                       QGauss<dim>(fe.degree + 2),
-                                      VectorTools::L2_norm);
+                                      VectorTools::L2_norm,
+                                      &select_interior_pressure);
 
     const double L2_error = difference_per_cell.l2_norm();
     std::cout << "L2_error_pressure " << L2_error << std::endl;
   }
 
 
 
-  // @sect4{WGDarcyEquation<dim>::postprocess}
-
-  // After we calculated the numerical pressure, we evaluate $L_2$ errors for
-  // the velocity on each cell and $L_2$ errors for the flux on faces.
+  // @sect4{WGDarcyEquation<dim>::compute_velocity_errors}
 
-  // We are going to evaluate velocities on each cell and calculate the
-  // difference between numerical and exact velocities. To calculate velocities,
-  // we need interior and face pressure values of each element, and some other
-  // cell matrices.
+  // In this function, we evaluate $L_2$ errors for the velocity on
+  // each cell, and $L_2$ errors for the flux on faces.
 
+  // We are going to evaluate velocities on each cell and calculate
+  // the difference between numerical and exact velocities. The
+  // velocity is defined as $\mathbf{u}_h = \mathbf{Q}_h \left(
+  // -\mathbf{K}\nabla_{w,d}p_h \right)$, which requires us to compute
+  // many of the same terms as in the assembly of the system matrix.
+  // There are also the matrices $E^K,D^K$ we need to assemble (see
+  // the introduction) but they really just follow the same kind of
+  // pattern.
+  //
+  // Computing the same matrices here as we have already done in the
+  // `assemble_system()` function is of course wasteful in terms of
+  // CPU time. Likewise, we copy some of the code from there to this
+  // function, and this is also generally a poor idea. A better
+  // implementation might provide for a function that encapsulates
+  // this duplicated code. One could also think of using the classic
+  // trade-off between computing efficiency and memory efficiency to
+  // only compute the $C^K$ matrices once per cell during the
+  // assembly, storing them somewhere on the side, and re-using them
+  // here.
   template <int dim>
   void WGDarcyEquation<dim>::compute_velocity_errors()
   {
     const FE_RaviartThomas<dim> fe_rt(0);
```

Comment:
Same remark here regarding the degree, `fe.degree`?

Reply:
Same as above with the separate PR.